### ISSUE TITLE
Tests fail with PyPy

### ISSUE THREAD

**Issue by steko** [totalopenstation/totalopenstation#41]
We have a small number of unit tests to help with our development. Today I ran the test suite with [PyPy](http://pypy.org/) and 3 of them failed, all related to the `PolarPoint` class in `formats/polar.py`.

```
[steko@ogma totalopenstation]$ pypy -m unittest discover
EEE...
======================================================================
ERROR: test_points (totalopenstation.tests.test_nikon.TestNikonParser)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/steko/code/totalopenstation/totalopenstation/tests/test_nikon.py", line 19, in test_points
    self.assertAlmostEqual(self.fp.points[0][1], 8.9092817528619808)
  File "/home/steko/code/totalopenstation/totalopenstation/formats/nikon_raw_v200.py", line 64, in _points
    points.append(p.to_point().tuplepoint)
  File "/home/steko/code/totalopenstation/totalopenstation/formats/polar.py", line 93, in to_point
    self.angle,
AttributeError: PolarPoint instance has no attribute 'angle'

======================================================================
ERROR: test_shorter (totalopenstation.tests.test_nikon.TestNikonParser)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/steko/code/totalopenstation/totalopenstation/tests/test_nikon.py", line 22, in test_shorter
    self.assertEqual(self.fp.points[2][4], 'P')
  File "/home/steko/code/totalopenstation/totalopenstation/formats/nikon_raw_v200.py", line 64, in _points
    points.append(p.to_point().tuplepoint)
  File "/home/steko/code/totalopenstation/totalopenstation/formats/polar.py", line 93, in to_point
    self.angle,
AttributeError: PolarPoint instance has no attribute 'angle'

======================================================================
ERROR: test_polar (totalopenstation.tests.test_polar.TestPolar)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/steko/code/totalopenstation/totalopenstation/tests/test_polar.py", line 21, in test_polar
    self.assertEqual(self.p0.to_cartesian(), '100.0,100.0,110.0')
  File "/home/steko/code/totalopenstation/totalopenstation/formats/polar.py", line 80, in to_cartesian
    self.angle,
AttributeError: PolarPoint instance has no attribute 'angle'

----------------------------------------------------------------------
Ran 6 tests in 0.001s

FAILED (errors=3)
```

---
- Bitbucket: https://bitbucket.org/steko/totalopenstation/issue/41
- Originally reported by: Stefano Costa
- Originally created at: 2013-03-28T18:43:57.356

**Comment by steko** [comment]
_From Stefano Costa on 2013-03-28 17:56:04+00:00_

The source of the error is in this code:

``` python

        if angle_type is 'deg':
            self.angle = math.radians(float(angle))
            self.z_angle = math.radians(float(z_angle))
        if angle_type is 'gon':
            self.angle = math.radians(float(angle)*0.9)
            self.z_angle = math.radians(float(z_angle)*0.9)
```

Using `is` is a bad idea, because CPython and PyPy don't behave in the same way.

``` pythonconsole
Python 2.7.2 (341e1e3821fff77db3bb5cdb7a4851626298c44e, Jul 21 2012, 19:41:17)
[PyPy 1.9.0 with GCC 4.7.1] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>>> from totalopenstation.formats.polar import BasePoint, PolarPoint
>>>> bp0 = BasePoint(x=100.0, y=100.0, z=100.0, ih=0)
>>>> p0 = PolarPoint(dist=10,
....                              angle=0,
....                              z_angle=0,
....                              th=0,
....                              angle_type='deg',
....                              base_point=bp0,
....                              pid=1,
....                              text='Test Point')
>>>> p0.angle_type
'deg'
>>>> p0.angle_type is 'deg'
False
>>>> p0.angle_type == 'deg'
True
```

```
Python 2.7.3 (default, Aug  9 2012, 17:23:57) 
[GCC 4.7.1 20120720 (Red Hat 4.7.1-5)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>>> from totalopenstation.formats.polar import BasePoint, PolarPoint
>>>> bp0 = BasePoint(x=100.0, y=100.0, z=100.0, ih=0)
>>>> p0 = PolarPoint(dist=10,
....                              angle=0,
....                              z_angle=0,
....                              th=0,
....                              angle_type='deg',
....                              base_point=bp0,
....                              pid=1,
....                              text='Test Point')
>>>> p0.angle_type
'deg'
>>>> p0.angle_type is 'deg'
True
>>>> p0.angle_type == 'deg'
True
```

CPython gives `True` with both `is` and `==` but PyPy only with `==`.

**Comment by steko** [comment]
_From Stefano Costa on 2013-03-28 18:01:03+00:00_

Don't assume identity between strings with the same value. Fix #41

→ <<cset 4b3f4dd0b69deaa7de0684ef3d97944ba0495665>>
